source/agora/common/Amount.d

```diff
@@ -169,6 +169,34 @@ public struct Amount
         return true;
     }
 
+    /***************************************************************************
+
+        Divide this `Amount` into `Q` equal parts and returns the remainder
+
+        This function allows to losslessly partition an `Amount`.
+        Usage example include distributing an `Amount` evenly among participant,
+        or computing an absolute majority of 2/3rd.
+
+        Params:
+            value = The value to divide by.
+                    Dividing by `0` is the same as dividing by `1`.
+
+        Returns:
+            The remainder of the division
+
+    ***************************************************************************/
+
+    pragma(inline, true)
+    public Amount div (uint value)
+    {
+        // No-op
+        if (value < 2) return Amount(0);
```

Comment:
Does the compiler recognize the `value` as the `value` of the parameter? Or should we recognize it as `this.value`?

Reply:
I see that the test code is working properly, so I know exactly what the value of the parameter is.
The compiler is smart.

Reply:
I should probably use a different name though, good point.